### PR TITLE
aws-lambda - removes the wrong assumption that headers are always set

### PR DESCRIPTION
**Pull request by hedefalk**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

Sorry, at the time, I can't find any good docs for this, better than https://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html

which states that "the event structure varies by service.".

However, the reason fot the PR is by imperical/anecdotal evidence :) When testing a lambda in aws console without providing any headers, null was passed giving my typechecked code a NPE.

### CODE REVIEW

File: types/aws-lambda/index.d.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ export interface APIGatewayEventRequestContext {
 // API Gateway "event"
 export interface APIGatewayProxyEvent {
     body: string | null;
-    headers: { [name: string]: string };
+    headers?: { [name: string]: string };
     multiValueHeaders: { [name: string]: string[] };
```
Comment (alex-bolenok-centralreach) [inline types/aws-lambda/index.d.ts]: I assume those are always set?

**Review by alex-bolenok-centralreach — APPROVED** [review]

**Comment by simonbuchan** [comment]
Hmm, making headers optional is a breaking change, I'm suspecting we're going to get complaints to change this back. I think they will never be non-null in practice, as you will always at least get the `Host` header from HTTP/1.1, so I would put this down as an unfortunate bug in the AWS console.

That said, it's not too hard a workaround, and this is already merged, so I guess wait to see if there are complaints?

**Comment by brianeletype** [comment]
> Hmm, making headers optional is a breaking change, I'm suspecting we're going to get complaints to change this back. I think they will never be non-null in practice, as you will always at least get the `Host` header from HTTP/1.1, so I would put this down as an unfortunate bug in the AWS console.
> 
> That said, it's not too hard a workaround, and this is already merged, so I guess wait to see if there are complaints?

I'll submit the first complaint.  This broke a bunch of code where we're getting the Host header.  Why would a breaking change like get through?

**Comment by simonbuchan** [comment]
@brianeletype and others, I've opened a revert MR 👆 in #40929

**Pull request merged**